velox/common/encode/Base64.cpp

```diff
@@ -483,15 +526,22 @@ std::string Base64::decodeUrl(folly::StringPiece encodedText) {
 void Base64::decodeUrl(
     const std::pair<const char*, int32_t>& payload,
     std::string& decodedOutput) {
-  size_t decodedSize = (payload.second + 3) / 4 * 3;
-  decodedOutput.resize(decodedSize, '\0');
-  decodedSize = Base64::decodeImpl(
+  size_t inputSize = payload.second;
+  auto decodedSize = calculateDecodedSize(payload.first, inputSize);
```

Comment:
Let's keep the current estimation `size_t decodedSize = (payload.second + 3) / 4 * 3;` and optimize this in a subsequent PR as discussed above.